pkg/grouper/package_grouper.go

```diff
@@ -0,0 +1,45 @@
+package grouper
+
+import (
+	"github.com/google/osv-scanner/internal/utility/purl"
+	"github.com/google/osv-scanner/pkg/models"
+	"slices"
+)
+
+func GroupByPURL(packageSources []models.PackageSource) map[string]models.PackageVulns {
```

Comment:
Can you add some function level comments explaining what this function is doing?

Reply:
Also side comment: `grouper` is likely not the right package for this.. @another-rex WDYT?

Reply:
Thanks for the review ! :) 
What package would you recommend ? (I thought of the `internal/utility/purl` package and then renaming the method to `Group`)

Reply:
thanks for addressing the comments! yep `internal/utility/purl` and renaming the method to `Group` sounds great to me!

Reply:
I'm on holiday right know, I will do it right when I come back (later this week)

Reply:
I moved the package there, I'm resolving the conversation, feel free to open it back if anything looks off ;)